### ISSUE TITLE
Delete command history suggestion

### ISSUE THREAD

**Issue by luis-pato** [warpdotdev/Warp#3675]
### Describe the solution you'd like?

```Text
A few weeks ago, I made a typo when entering a command.
Now, I continually receive this incorrect command as an autocomplete suggestion, which has caused me to accidentally run it multiple times.

I believe it would be beneficial to have a feature that allows us to remove autocomplete suggestions from our command history. Perhaps, when the autocomplete suggestion list appears as we start typing, we could include a "Remove suggestion" button when hovering over one of the suggestions.
```

### How important is this feature to you?

2

### Warp Internal (ignore) - linear-label:770f6576-d6c0-4e4f-a259-fc64b5156087

None

**Comment by dannyneira** [comment]
@luis-pato As of last week's Warp release `v0.2023.09.12.16.17` or newer, Autosuggestions now prioritize commands from history that were executed in the current directory so that should help. If you're still seeing the issue. As a possible workaround, you can delete the entry from `history` to prevent it from showing up in autosuggestions (i.e. `history -d {{line_number}}`)

**Comment by luis-pato** [comment]
thanks @dannyneira that sounds like an interesting tip.

Unfortunately it didn't work.
I have done `history` and then searched for the line i wanted to delete.
Then i have done for instance `history -d 11594`. I get back a new history list. When i search it i still see the line 11594 and when i try out the command i also still see the incorrect suggestion (but you are right, the "correct suggestion" is suggested first).
I need to look around to see why it isn't working...

**Comment by Knogobert** [comment]
Firefox (& other browsers) have this removal functionality in the omni-bar, by entering ctrl+shift+backspace. It felt natural to me to try it in warp as well. Perhaps at least add this as a shortcut and you don’t have to worry about the UI that much for starters. To try and see if this suffices for us plebs <3

**Comment by pete-willard** [comment]
The ability to shortcut-delete a highlighted suggestion would be a most welcome one. Bump!

**Comment by dannypernik** [comment]
Yes please. I keep getting the same suggestion for an alias I deleted months ago

**Comment by mcshaman** [comment]
Such a big bugbear for me

**Comment by dannypernik** [comment]
> Hey y'all - trying to take this on in my spare time :) can't guarantee a ship date but hoping to have something out relatively soon. A little tricky to get it working in all shells we support but will do my best! Small demo below.
> 
>  Jul-18-2025.18-21-59.mp4

Yes! This is precisely what is needed. Would it delete all prior exact matches from history?

**Comment by ApplePieGiraffe** [comment]
Hey any update on this? I've made so many typos when running commands most of my most common suggestions are frequently incorrect :(

**Comment by martinliptak** [comment]
How can I delete the credential? I opened "Library/Application Support/dev.warp.Warp-Stable/warp.sqlite" and its showing me that there're no entities.

**Comment by BassemWaheed44** [comment]
I have just asked the AI Agent on Warp site, and is said 
> Your shell history file (e.g. ~/.zsh_history)

So I opened `~/.zsh_history` and deleted the exact command I wanted to remove!! lmk if it works with you!

**Comment by clementaudic** [comment]
Is this feature still present? On my side, there are no such `x` buttons in the command history of Warp v0.2026.08.26.17.59.stable_01.

**Comment by jefflloyd** [comment]
@clementaudic do you have the X enabled? It's in Settings > Features > Terminal Input > "Show autosuggestion ignore button"
